### PR TITLE
Add both CRD-backed NP and K8s-backed NP to the NP watcher

### PR DESCRIPTION
**Pull request by robbrockbank**
Implements the KDD NP watch by including both K8s NP and CRD-backed NP in the watcher.
Also adds resource version to the CRD list function.

### CODE REVIEW

File: lib/backend/k8s/resources/networkpolicy.go [hunk 1]
```diff
@@ -170,58 +171,82 @@ func (c *networkPolicyClient) List(ctx context.Context, list model.ListInterface
...
 		Get().
-		Resource("networkpolicies").
-		Timeout(10 * time.Second).
-		Do().Into(&networkPolicies)
+		Resource("networkpolicies")
+	if l.Namespace != "" {
+		// Add the namespace if requested.
+		req = req.Namespace(l.Namespace)
+	}
+	log.Warning("Sending query")
```
Comment (doublek) [inline lib/backend/k8s/resources/networkpolicy.go]: Remove this log.

File: lib/backend/k8s/resources/networkpolicy.go [hunk 2]
```diff
@@ -170,58 +171,82 @@ func (c *networkPolicyClient) List(ctx context.Context, list model.ListInterface
...
 	}
 
+	// If a revision is specified, see if it contains a "/" and if so split into separate
+	// revisions for the CRD and for the K8s resource.
+	k8sRev := revision
+	crdRev := revision
+	if strings.Contains(revision, "/") {
+		revs := strings.Split(revision, "/")
+		if len(revs) != 2 {
+			return nil, fmt.Errorf("badly formatter ResourceVersion: %s", revision)
```
Comment (doublek) [inline lib/backend/k8s/resources/networkpolicy.go]: nit: s/formatter/formatted

File: lib/backend/k8s/resources/networkpolicy.go (hunk 1 above)
Comment (robbrockbank) [reply lib/backend/k8s/resources/networkpolicy.go]: lal alalalala

File: lib/backend/k8s/resources/networkpolicy.go [hunk 3]
```diff
@@ -232,6 +257,109 @@ func (c *networkPolicyClient) Watch(ctx context.Context, list model.ListInterfac
...
+}
+
+func newNetworkPolicyWatcher(ctx context.Context, k8sWatch, calicoWatch api.WatchInterface) api.WatchInterface {
+	ctx, cancel := context.WithCancel(ctx)
+	wc := &networkPolicyWatcher{
+		k8sNPWatch:    k8sWatch,
+		calicoNPWatch: calicoWatch,
+		context:       ctx,
+		cancel:        cancel,
+		resultChan:    make(chan api.WatchEvent, resultsBufSize),
```
[the remark this answers is not in the document]
Comment (robbrockbank) [reply lib/backend/k8s/resources/networkpolicy.go]: completely arbitrary

the syncer watcher has a bigger buffer - so I don't think this one needs to be large.

**Pull request merged**